spec/lib/middleman-cdn/cdns/cloudflare_spec.rb

```diff
@@ -73,20 +96,45 @@
         expect { subject.invalidate(options, files) }.to output(/✔/).to_stdout
       end
 
-      context "matches all files" do
+      context "max 50 files to invalidate" do
         before do
-          allow(double_cloudflare).to receive(:fpurge_ts)
+          expect(files.count).to be <= 50
         end
 
-        it "should invalidate the entire zone" do
-          expect(double_cloudflare).to receive(:fpurge_ts).once.with("example.com")
-          subject.invalidate(options, files, all: true)
+        it_behaves_like "invalidating individual files"
+
+        it "should call cloudflare to purge each file for each base url" do
+          files.each do |file|
+            expect(double_cloudflare).to receive(:zone_file_purge).once.ordered.with("example.com", "http://example.com#{file}")
+          end
+          subject.invalidate(options, files)
         end
+      end
 
-        it "should not invalidate individual files" do
-          expect(double_cloudflare).not_to receive(:zone_file_purge)
-          subject.invalidate(options, files, all: true)
+      context "more than 50 files to invalidate" do
+        let(:files) { super() + ["/index.html"] }
```

Comment:
:smile: 
I had to look it up myself: https://github.com/rspec/rspec-core/commit/c5172be7378fcadf2b468c6d0ed367bb7471985c#diff-112a3ec80f232a6de7289060604874f6R39